### PR TITLE
50 dev

### PR DESCRIPTION
**Pull request by JahodaPaul**
## Describe your changes
Developed algorithm for optimal sampling of architecture for number of local optima estimation to save computational power (As described by Andres)

## What is the related task number?
50

## Checklist before requesting a review
- [X] I have performed a self-review of my code
- [X] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: src/tum_dlr_automl_for_eo/utils/custom_metrics.py [hunk 1]
```diff
@@ -90,6 +90,48 @@ def perform_bils_algorithm(architectures, acc_list, starting_point, num_of_nei_t
...
 
+
+def get_hamming_distance(architecture_1, architecture_2):
+    return np.count_nonzero(architecture_1 != architecture_2)
+
+"""
+find starting points with high density of fully trained architectures, while simultaneously avoiding sampling twice
+from the same area 
+"""
+def find_k_starting_points(architectures, k,radius=3):
```
Comment (emreds) [inline src/tum_dlr_automl_for_eo/utils/custom_metrics.py:101]: So here we assume that architectures are already binary encoded right?

Comment (Stargazer1998) [reply src/tum_dlr_automl_for_eo/utils/custom_metrics.py:101]: Yes, that is the assumption

**Comment by emreds** [comment]
@JahodaPaul Hi Pavel! Should I merge this branch or the changes were included in another branch?

**Comment by JahodaPaul** [comment]
> @JahodaPaul Hi Pavel! Should I merge this branch or the changes were included in another branch?

Merge I think

**Pull request closed without merging**